code/modules/projectiles/guns/magic/wand.dm

```diff
@@ -140,6 +140,15 @@
 
 /obj/item/gun/magic/wand/polymorph/zap_self(mob/living/user)
 	..() //because the user mob ceases to exists by the time wabbajack fully resolves
+
+	// Disconnect AI's in shells
+	if(istype(user, /mob/living/silicon/robot))

```

Comment:
I am pretty sure you just want to use `undeploy()` here, or `disconnect_shell()` on the AI mob. This code alone does not look like it would actually return the AI as it is currently.

Reply:
I changed this and moved where it happens.
Good eye.